libraries/Microsoft.Bot.Builder.Skills/Dialogs/SkillDialog.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Builder.Skills.Dialogs;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// A sample dialog that can wrap remote calls to a skill.
+    /// </summary>
+    /// <remarks>
+    /// The options parameter in <see cref="BeginDialogAsync"/> must be a <see cref="SkillDialogArgs"/> instance
+    /// with the initial parameters for the dialog.
+    /// </remarks>
+    public class SkillDialog : Dialog
+    {
+        private readonly IStatePropertyAccessor<BotFrameworkSkill> _activeSkillProperty;
+        private readonly ConversationState _conversationState;
+        private readonly SkillDialogOptions _dialogOptions;
+
+        public SkillDialog(SkillDialogOptions dialogOptions, ConversationState conversationState)
+            : base(nameof(SkillDialog))
+        {
+            _dialogOptions = dialogOptions ?? throw new ArgumentNullException(nameof(dialogOptions));
+            _conversationState = conversationState ?? throw new ArgumentNullException(nameof(conversationState));
+            _activeSkillProperty = conversationState.CreateProperty<BotFrameworkSkill>($"{typeof(SkillDialog).FullName}.ActiveSkillProperty");
+        }
+
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default)
+        {
+            if (!(options is SkillDialogArgs dialogArgs))
```

Comment:
This check seems a little odd. I think it technically throws an exception for all the cases, however, it sometimes throws a null exception for a something that isn't null which seems a little confusing. It would be OK to have two exceptions... if this is required then first check for null (and throw ArgumentNull) and then check for type.

Reply:
Move this code to a separate validation helper.